bundle/Command/BaseMultiprocessCommand.php

```diff
@@ -0,0 +1,390 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Netgen\Bundle\SiteBundle\Command;
+
+use Generator;
+use Netgen\Bundle\SiteBundle\Command\MultiprocessCommand\Items;
+use Psr\Log\LoggerInterface;
+use RuntimeException;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Exception\InvalidOptionException;
+use Symfony\Component\Console\Helper\ProgressBar;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\OutputStyle;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\Process\PhpExecutableFinder;
+use Symfony\Component\Process\Process;
+
+use function array_fill;
+use function ceil;
+use function count;
+use function explode;
+use function fgets;
+use function file_get_contents;
+use function implode;
+use function in_array;
+use function is_file;
+use function is_numeric;
+use function min;
+use function pclose;
+use function popen;
+use function preg_match;
+use function preg_match_all;
+use function spl_object_id;
+use function sprintf;
+use function stream_get_contents;
+use function usleep;
+
+use const DIRECTORY_SEPARATOR;
+
+abstract class BaseMultiprocessCommand extends Command
+{
+    protected ?string $phpPath;
+    protected string $projectDir;
+    protected LoggerInterface $logger;
+    protected InputInterface $input;
+    protected OutputStyle $outputStyle;
+    protected ProgressBar $progressBar;
+
+    public function __construct(
+        string $projectDir,
+        LoggerInterface $logger,
+        ?string $phpPath = null
+    ) {
+        $this->projectDir = $projectDir;
+        $this->logger = $logger;
+        $this->phpPath = $phpPath;
+
+        parent::__construct();
+    }
+
+    abstract protected function getCount(): int;
+
+    abstract protected function getItemsGenerator(int $limit): Generator;
+
+    abstract protected function process($item): void;
+
+    protected function configure(): void
+    {
+        $this->addOption(
+            'limit',
+            null,
+            InputOption::VALUE_OPTIONAL,
+            'Number of items to process in a single iteration',
+            50,
+        );
+
+        $this->addOption(
+            'processes',
+            null,
+            InputOption::VALUE_OPTIONAL,
+            'Maximum number of processes to run in parallel. By default it will use the number of CPU cores minus 1.',
+            'auto',
+        );
+
+        $this->addOption(
+            'items',
+            null,
+            InputOption::VALUE_OPTIONAL,
+            'Comma-separated list of item identifiers',
+        );
+    }
+
+    protected function initialize(InputInterface $input, OutputInterface $output): void
+    {
+        $this->input = $input;
+        $this->outputStyle = new SymfonyStyle($input, $output);
```

Comment:
This is not strictly `outputStyle` as it also defines methods to ask user for data (`ask`, `confirm` and so on)

Reply:
Fixed in https://github.com/netgen/site-bundle/pull/30/commits/2654801d07bbda83be57e954a9afa53566fa1ea8